### ISSUE TITLE
"Replacement transaction underpriced" even though the difference is >10%

### ISSUE THREAD

**Issue by CherryDT** [hyperledger/besu-docs#889]
### Description
As a user I'd like to be able to submit replacement transactions with 18% higher gas

### Steps to Reproduce (Bug)
1. Submit eip1559 transaction with maxFeePerGas of 110 gwei
2. Submit eip1559 transaction with maxFeePerGas of 130 gwei (which is 18% higher) with same nonce
3. Observe error "Replacement transaction is underpriced"

The docs say that the default of the min. bump would be 10% and I never changed that.

Note the same thing works with fine with an OpenEthereum node.

**Expected behavior:** Transaction should be accepted

**Actual behavior:** Error "Replacement transaction is underpriced"

**Frequency:** On every fee bump attempt

### Versions (Add all that apply)
* Software version: besu/v21.10.2/linux-x86_64/openjdk-java-11
* Java version: openjdk 11.0.11 2021-04-20; OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04); OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)
* OS Name & Version: Ubuntu 20.04.2 LTS
* Kernel Version: Linux ip-172-31-4-149 5.11.0-1020-aws `#21`~20.04.2-Ubuntu SMP Fri Oct 1 13:03:59 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
* Cloud VM, type, size: AWS t3a.xlarge

**Comment by garyschulteog** [comment]
> ...
>
> https://docs.alchemy.com/alchemy/guides/eip-1559/maxpriorityfeepergas-vs-maxfeepergas
> 
> Is this article wrong?

I am on mobile, but skimming that article, it looks correct but doesn't address transaction replacement post-london.  AFAIK all London compatible clients implement a similar mechanism for transaction replacement based on the "effective" priority fee:

 https://github.com/ethereum/go-ethereum/issues/23616#issuecomment-924657965

https://githubmemory.com/repo/openethereum/openethereum/issues/506

We could probably stand to explicitly document beau's post-london transaction replacement behavior if we don't already.

**Comment by garyschulte** [comment]
TL/DR: I think this *is* actually a bug, and also an opportunity for documentation.

First off, Besu's replacement implementation differs from [go-ethereum's impelmentation](https://github.com/ethereum/go-ethereum/blob/master/core/tx_list.go#L301-L303) which requires both feeCap (maxFeePerGas) and tip (priorityFee) to be bumped by a configured value (default 10%).

Besu's implementation is somewhat more complex than just bumping effective priority by 10%.  Besu, post-london has two transaction replacement rules that determine whether a transaction is replaced.  Both default to 'price bump' configuration of 10%:
* [legacy transaction replaces a legacy transaction rule](https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionReplacementByGasPriceRule.java), if new gasPrice is >= 110% of prior gasPrice, replace

* [1559 transaction replacement rule](https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionReplacementByFeeMarketRule.java).  This is more complex since it deals with 1559 replacing legacy transactions and legacy transaction replacing 1559 transactions.  Post-london, every transaction has an implicit effective priority fee.  Legacy transactions' effective priority fee is gasPrice - current base fee.  1559 transactions are the minimum of (maxFeePerGas - baseFee) or (maxPriorityFeePerGas).  

The effective price of a transaction is baseFee + effective priority fee

Transactions are replaced when either :
1. [the effective price is bumped by 10% AND new effective priority fee is the same or greater than the prior tx](https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionReplacementByFeeMarketRule.java#L65-L68)
2. [the effective price is the same as the prior price and the priority fee is bumped by 10%](https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionReplacementByFeeMarketRule.java#L69-L74)

Reading through the implementation again however I believe the second condition on line 69 should be expanded to cover when the effective price is the same or *greater than* the prior transaction, and I believe this is the case you are encountering where the transaction is not getting replaced.
